### PR TITLE
pkg/trace/api: fix an edge-case in tag normalization

### PR DESCRIPTION
**Pull request by gbbr**
This change fixes an edge-case in tag normalization where unicode
replacement characters were treated as invalid UTF-8.

### CODE REVIEW

File: pkg/trace/api/normalizer_test.go [hunk 1]
```diff
@@ -430,6 +431,7 @@ func TestNormalizeTag(t *testing.T) {
 		{in: "test\x99\x8faaa", out: "test_aaa"},
 		{in: "test\x99\x8f", out: "test"},
 		{in: strings.Repeat("a", 888), out: strings.Repeat("a", 200)},
+		{in: strings.Repeat("a", 888), out: strings.Repeat("a", 200)},
```
Comment (AlexJF) [inline pkg/trace/api/normalizer_test.go]: Looks like a duplicate test case.

File: pkg/trace/api/normalizer_test.go [hunk 2]
```diff
@@ -443,6 +445,8 @@ func TestNormalizeTag(t *testing.T) {
 			}(),
 			out: "a", // 'b' should have been truncated
 		},
+		{"a" + string(unicode.ReplacementChar), "a"},
+		{"a" + string(unicode.ReplacementChar) + string(unicode.ReplacementChar), "a"},
```
Comment (AlexJF) [inline pkg/trace/api/normalizer_test.go]: A better test case would be to have valid letters after a replacement char and ensure the output contains them no?

File: pkg/trace/api/normalizer_test.go (hunk 1 above)
Comment (gbbr) [reply pkg/trace/api/normalizer_test.go]: I reviewed my own PR and yet I've missed this. Thanks. Removed.

File: pkg/trace/api/normalizer_test.go (hunk 2 above)
Comment (gbbr) [reply pkg/trace/api/normalizer_test.go]: Indeed, I added it.

**Review by AlexJF — APPROVED** [review]

**Pull request merged**